### PR TITLE
Move integration test suite from molecule to ansible-test

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

``molecule`` has been replaced with ``ansible-test``
some test cases have been updated 

- ``k8s_apply`` : remove duplicated tasks increasing the running time of the test
- ``helm``: use different namespaces for different test cases in order to wait for the namespace deletion before moving to the next test.
- ``all``: remove ``wait: yes`` at the end of each test when deleting namespace, the role used to create namespace will ensure that it is deleted before if existing.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

integration testing

### CODE REVIEW

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Review by gravesm — COMMENTED** [review]
This is great @abikouo! One small comment, but otherwise looks good. Thanks for doing this.

**Review by goneri — APPROVED** [review]

File: tests/integration/targets/k8s_apply/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+- block:
+    - name: Ensure namespace exists
+      k8s:
+        definition:
+          apiVersion: v1
+          kind: Namespace
+          metadata:
+            name: "{{ test_namespace }}"
+
+    # - name: Add a configmap
```
[the remark this answers is not in the document]
Comment (abikouo) [reply tests/integration/targets/k8s_apply/tasks/main.yml:11]: good catch!! That's commented during my debugging, there were a lot of duplicated cases in these files increasing the execution time.

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Review by gravesm — APPROVED** [review]
LGTM once we can actually get CI to pass.

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
regate

File: tests/integration/targets/k8s_apply/tasks/main.yml [hunk 2]
```diff
@@ -449,326 +450,14 @@
         that:
           - deploy_after_serviceaccount_removal is failed
 
-    - name: Insert new service port
```
Comment (gravesm) [inline tests/integration/targets/k8s_apply/tasks/main.yml:452]: Why did these get deleted?

Comment (abikouo) [reply tests/integration/targets/k8s_apply/tasks/main.yml:452]: all this is a duplicate test starting at line 206, seems like a wrong copy/paste

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
regate

**Pull request merged**